plugins/net_plugin/net_plugin.cpp

```diff
@@ -654,18 +661,11 @@ namespace eosio {
 
       void fill_out_buffer( std::vector<boost::asio::const_buffer>& bufs ) {
          fc::lock_guard g( _mtx );
-         if( !_sync_write_queue.empty() ) { // always send msgs from sync_write_queue first
-            fill_out_buffer( bufs, _sync_write_queue );
-         } else { // postpone real_time write_queue if sync queue is not empty
+         if( !_write_queue.empty() ) { // always send msgs from write_queue first
             fill_out_buffer( bufs, _write_queue );
-            EOS_ASSERT( _write_queue_size == 0, plugin_exception, "write queue size expected to be zero" );
-         }
-      }
-
-      void out_callback( boost::system::error_code ec, std::size_t w ) {
-         fc::lock_guard g( _mtx );
-         for( auto& m : _out_queue ) {
-            m.callback( ec, w );
+         } else {
+            fill_out_buffer( bufs, _trx_write_queue );
+            assert(_trx_write_queue.empty() && _write_queue.empty() && _write_queue_size == 0);
```

Comment:
Why `_trx_write_queue.empty()`?

Reply:
`_write_queue.empty` because we would have gone into the `if` instead of the `else` if `_write_queue` not empty.
`_trx_write_queue.empty()` because we just filled `bufs` from it, which copies everything out.